File: library/src/main/java/com/zdg/stateviewmanager/mananger/IStateViewManager.java
package com.zdg.stateviewmanager.mananger;

import android.view.View;
import android.view.ViewGroup;

import com.zdg.stateviewmanager.creator.StateLoader;

public interface IStateViewManager extends StateViewChanger, StateLoader<View> {


    ViewGroup getOverallView();


    /**
     * 设置核心布局
     *
     * @param resId
     * @return
     */
    View setContentView(int resId);

    View setContentView(View view);


    View getContentView();
}

File: stateview/src/main/java/com/zdg/stateviewmanager/mananger/IStateViewManager.kt
package com.zdg.stateviewmanager.mananger

import android.view.View
import android.view.ViewGroup

import com.zdg.stateviewmanager.creator.StateLoader

interface IStateViewManager : StateViewChanger, StateLoader<View> {


    val overallView: ViewGroup?


    val contentView: View?


    /**
     * 设置核心布局
     *
     * @param resId
     * @return
     */
    fun setContentView(resId: Int): View

    fun setContentView(view: View): View
}
